WebsocketClient.nonBlocking cannot receive messages in binary mode

The non-blocking websocket client does not seem to receive binary mode messages.  Send works fine.

This failing test is excerpted from `WebsocketClientTest`, so is missing some of the shared setup code.

```kotlin
val ws = websockets(
    "/bin" bind { ws: Websocket ->
        ws.onMessage {
            val content = it.body.stream.readBytes()
            ws.send(WsMessage(content.inputStream()))
            ws.close(NORMAL)
        }
    }
)

@Test
fun `non-blocking - send and receive in binary mode`() {
    val queue = LinkedBlockingQueue<() -> WsMessage?>()
    val received = generateSequence { queue.take()() }

    WebsocketClient.nonBlocking(Uri.of("ws://localhost:$port/bin")) { ws ->
        ws.onMessage { message ->
            queue.add { message }
        }
        ws.send(WsMessage("hello".byteInputStream()))
        ws.onClose {
            queue.add { null }
        }
    }

    val messages = received.take(4).toList()
    assertThat(messages, hasSize(equalTo(1)))
    assertThat(messages.first().body is StreamBody, equalTo(true))
    assertThat(messages.first().bodyString(), equalTo("hello"))
}
```

> expected: a value that has size that is equal to 1
but had size that was: 0
java.lang.AssertionError: expected: a value that has size that is equal to 1
but had size that was: 0

--- a/http4k-client/websocket/src/main/kotlin/org/http4k/client/internal/internal.kt
+++ b/http4k-client/websocket/src/main/kotlin/org/http4k/client/internal/internal.kt
@@ -57,6 +57,10 @@ class NonBlockingClient(uri: Uri, headers: Headers, timeout: Duration, private v
 
     override fun onMessage(message: String) = socket.get().triggerMessage(WsMessage(message))
 
+    override fun onMessage(bytes: ByteBuffer) {
+        socket.get().triggerMessage(WsMessage(Body(bytes)))
+    }
+
     override fun onError(e: Exception) = socket.get().triggerError(e)
 }
 

--- a/http4k-client/websocket/src/main/kotlin/org/http4k/client/internal/internal.kt
+++ b/http4k-client/websocket/src/main/kotlin/org/http4k/client/internal/internal.kt
@@ -57,9 +57,7 @@ class NonBlockingClient(uri: Uri, headers: Headers, timeout: Duration, private v
 
     override fun onMessage(message: String) = socket.get().triggerMessage(WsMessage(message))
 
-    override fun onMessage(bytes: ByteBuffer) {
-        socket.get().triggerMessage(WsMessage(Body(bytes)))
-    }
+    override fun onMessage(bytes: ByteBuffer) = socket.get().triggerMessage(WsMessage(Body(bytes.array().inputStream())))
 
     override fun onError(e: Exception) = socket.get().triggerError(e)
 }

--- a/http4k-client/websocket/src/test/kotlin/org/http4k/client/WebsocketClientTest.kt
+++ b/http4k-client/websocket/src/test/kotlin/org/http4k/client/WebsocketClientTest.kt
@@ -30,6 +30,14 @@ class WebsocketClientTest {
     @BeforeEach
     fun before() {
         val ws = websockets(
+            "/bin" bind { ws: Websocket ->
+                ws.onMessage {
+                    val content = it.body.stream.readBytes()
+                    ws.send(WsMessage(content.inputStream()))
+                    ws.close(NORMAL)
+                }
+            },
+
             "/{name}" bind { ws: Websocket ->
                 val name = ws.upgradeRequest.path("name")!!
                 ws.send(WsMessage(name))
@@ -134,5 +142,23 @@ class WebsocketClientTest {
 
         assertThat(received.take(4).toList(), equalTo(listOf(WsMessage("bob"), WsMessage("hello"))))
         assertThat(connected, equalTo(true))
+    }
+
+    @Test
+    fun `non-blocking - send and receive in binary mode`() {
+        val queue = LinkedBlockingQueue<() -> WsMessage?>()
+        val received = generateSequence { queue.take()() }
+
+        WebsocketClient.nonBlocking(Uri.of("ws://localhost:$port/bin")) { ws ->
+            ws.onMessage { message ->
+                queue.add { message }
+            }
+            ws.send(WsMessage("hello".byteInputStream()))
+            ws.onClose {
+                queue.add { null }
+            }
+        }
+
+        assertThat(received.take(4).toList(), equalTo(listOf(WsMessage("hello"))))
     }
 }

--- a/http4k-client/websocket/src/test/kotlin/org/http4k/client/WebsocketClientTest.kt
+++ b/http4k-client/websocket/src/test/kotlin/org/http4k/client/WebsocketClientTest.kt
@@ -2,6 +2,8 @@ package org.http4k.client
 
 import com.natpryce.hamkrest.assertion.assertThat
 import com.natpryce.hamkrest.equalTo
+import com.natpryce.hamkrest.hasSize
+import org.http4k.core.StreamBody
 import org.http4k.core.Uri
 import org.http4k.routing.bind
 import org.http4k.routing.path
@@ -153,12 +155,14 @@ class WebsocketClientTest {
             ws.onMessage { message ->
                 queue.add { message }
             }
-            ws.send(WsMessage("hello".byteInputStream()))
             ws.onClose {
                 queue.add { null }
             }
+            ws.send(WsMessage("hello".byteInputStream()))
         }
 
-        assertThat(received.take(4).toList(), equalTo(listOf(WsMessage("hello"))))
+        val messages = received.take(4).toList()
+        assertThat(messages.all { it.body is StreamBody }, equalTo(true))
+        assertThat(messages, equalTo(listOf(WsMessage("hello"))))
     }
 }

